app/models/metric/capture.rb

```diff
@@ -31,210 +31,48 @@ def self.concurrent_requests(interval_name)
+    Array.wrap(ems).each do |e|
+      MiqQueue.put_unless_exists(
+        :class_name  => "Metric::Capture",
+        :method_name => "perf_collect_all_metrics",
+        :args        => [e.id, interval_name, start_time, end_time, options],
+        :role        => "ems_metrics_collector",
+        :queue_name  => e.metrics_collector_queue_name,
+        :zone        => e.zone_id,
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :state       => %w(ready dequeue),
       )
-      _log.info("Created task id: [#{task.id}] for: [#{pkey}] with targets: #{targets_by_rollup_parent[pkey].inspect} for time range: [#{task_start_time} - #{task_end_time}]")
-      targets.each do |target|
-        h[target] = {
-          :task_id => task.id,
-          :force   => true, # Force collection since we've already verified that capture should be done now
-          :zone    => zone,
-        }
-      end
     end
   end
-  private_class_method :calc_target_options
-
-  def self.queue_captures(targets, target_options)
-    # Queue the captures for each target
-    use_historical = historical_days != 0
-
-    targets.each do |target|
-      interval_name = perf_target_to_interval_name(target)
-
-      options = target_options[target]
 
-      begin
-        target.perf_capture_queue(interval_name, options)
-        if !target.kind_of?(Storage) && use_historical && target.last_perf_capture_on.nil?
-          target.perf_capture_queue('historical')
-        end
-      rescue => err
-        _log.warn("Failed to queue perf_capture for target [#{target.class.name}], [#{target.id}], [#{target.name}]: #{err}")
-      end
-    end
+  def self.perf_collect_all_metrics(ems, interval_name = "realtime", start_time = nil, end_time = nil, options = {})
+    ems = ExtManagementSystem.find(ems) unless ems.kind_of?(ExtManagementSystem)
+    klass = ems.class::MetricsCapture
+    klass.new(nil, ems).perf_collect_all_metrics(interval_name, start_time, end_time, options)
   end
-  private_class_method :queue_captures
 
-  def self.perf_target_to_interval_name(target)
-    case target
-    when Host, VmOrTemplate then                       "realtime"
-    when ContainerNode, Container, ContainerGroup then "realtime"
-    when Storage then                                  "hourly"
-    end
+  def self.perf_capture_gap(start_time, end_time, zone_id = nil)
+    perf_capture_gap_queue(start_time, end_time, zone_id)
   end
-  private_class_method :perf_target_to_interval_name
 
-  def self.minutes_ago(value)
-    if value.kind_of?(Integer) # Default unit is minutes
-      value.minutes.ago.utc
-    elsif value.nil?
-      nil
-    else
-      value.to_i_with_method.seconds.ago.utc
-    end
+  # called by ui
+  # run a perf capture zone for a zone or ems
+  def self.perf_capture_gap_queue(start_time, end_time, zone = nil)
+    emses = if zone.kind_of?(ExtManagementSystem)
```

Comment:
Can we refactor this method's signature? The way this param `zone` can be an `ems` is against intuition.